server/util/grpc_client/grpc_client.go

```diff
@@ -6,16 +6,28 @@ import (
 	"net/url"
 	"time"
 
+	"github.com/buildbuddy-io/buildbuddy/server/environment"
 	"github.com/buildbuddy-io/buildbuddy/server/rpc/interceptors"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials/google"
 	"google.golang.org/grpc/keepalive"
 )
 
-// DialTarget handles some of the logic around detecting the correct GRPC
+// DialSimple handles some of the logic around detecting the correct GRPC
 // connection type and applying relevant options when connecting.
-func DialTarget(target string, extraOptions ...grpc.DialOption) (*grpc.ClientConn, error) {
-	dialOptions := CommonGRPCClientOptions()
+//
+// NOTE: Clients within BuildBuddy servers (i.e. executor) should use the DialInterservice
+// variant with an env argument.
+func DialSimple(target string, extraOptions ...grpc.DialOption) (*grpc.ClientConn, error) {
+	return DialInterservice(nil, target, extraOptions...)
```

Comment:
passing nil as env (and then passing it all the way down to getClientIdentityAdder()) feels a little bit error prone to me. If the major difference between these two is whether or not the client identity interceptor is installed -- could we instead go back to having a

dialOptions := CommonGRPCClientOptions()

and then appending one more interceptor to it if env is set?

Reply:
done